livox_ros_driver/launch/livox_lidar_nodelet.launch

```diff
@@ -0,0 +1,34 @@
+<launch>
+
+	<arg name="manager_name" default="livox_manager"/>
+
+	<arg name="lvx_file_path" default="livox_test.lvx"/>
+	<arg name="bd_list" default="100000000000000"/>
+	<arg name="xfer_format" default="0"/>
+	<arg name="multi_topic" default="0"/>
+	<arg name="data_src" default="0"/>
+	<arg name="publish_freq" default="10.0"/>
+	<arg name="output_type" default="0"/>
+	<arg name="cmdline_arg" default="$(arg bd_list)"/>
+	<arg name="lidar_msg_frame_id" default="livox_frame"/>
+	<arg name="imu_msg_frame_id" default="livox_frame"/>
+	<arg name="lidar_bag" default="true"/>
+	<arg name="imu_bag" default="true"/>
+
+	<node name="$(arg manager_name)" pkg="nodelet" type="nodelet" args="manager"  output="screen"/>
```

Comment:
What if we wanted to use an external nodelet? Will this work if we define a nodelet manager in the roslaunch pipleine and then pass in the name to this launch file?

Reply:
Good question. That's probably a change that we should do later. We have several nodelets that could run in this nodelet manager so we should probably define a nodelet for the Lidar and all the other nodes should receive the same. Or we could create a new launch file calling all the nodelets nodes in one place.

Reply:
ok, I've been prototyping support for this:

```
<launch>
    <!-- Launch arguments. -->
    <arg name="nodelet" default="true"/>
    <arg name="use_external_manager" default="false"/>
    <arg name="nodelet_manager_name" default="strainer_detection_manager" />

    <!-- In order to reuse the nodes with and without nodelets, these variables
         are created. -->
    <arg name="package" value="strainer_detection" unless="$(arg nodelet)"/>
    <arg name="package" value="nodelet" if="$(arg nodelet)"/>
    <arg name="package_type" value="strainer_detection_node" unless="$(arg nodelet)"/>
    <arg name="package_type" value="nodelet" if="$(arg nodelet)"/>

    <!-- Nodelet manager. -->
    <group if="$(arg nodelet)">
        <!-- Only create a manager if not using an external one. -->
        <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager"
            unless="$(arg use_external_manager)"/>
    </group>

    <!-- Node arguments. -->
    <arg name="input_topic" default="input"/>

    <!-- Node. -->
    <node pkg="$(arg package)" type="$(arg package_type)" name="strainer_detection" output="screen"
          args="load strainer_detection/StrainerDetection $(arg nodelet_manager_name)">
        <rosparam file="$(find strainer_detection)/config/strainer_detection.yaml" command="load" />
        <remap from="~input" to="$(arg input_topic)"/>
    </node>
</launch>
```

Reply:
Do we really need something like this? We already know beforehand where is the nodelet. I'd only need a way to run with our without nodelets.